### ISSUE TITLE
[Question] Results vastly different for an agent created with Stable Baselines3 using hyperparameters optimized in RL Baselines3 Zoo.

### ISSUE THREAD

**Issue by mzelazko** [DLR-RM/rl-baselines3-zoo#458]
### ❓ Question

Hello,
I first optimize A2C on 1mln steps using RL Baselines3 Zoo:
 ```
python -m train --algo a2c --env ALE/Seaquest-ram-v5 -n 1000000 -optimize --n-trials 100 --n-startup-trials 10
--sampler tpe --pruner median  --n-evaluations 4 --n-eval-envs 16 --storage "some_valid_database" --study-name test
```
Top 3 results:
![mysqlsh_ugrHTRYZRL](https://github.com/DLR-RM/rl-baselines3-zoo/assets/148492076/56c85d96-9786-467b-b371-49978a3a3fb3)
Then using for example these hyperparameters:
![mysqlsh_4UsJxMM74z](https://github.com/DLR-RM/rl-baselines3-zoo/assets/148492076/251c9bcc-6bf0-437c-bc92-56388ea49f50)
and using this code:
```
def linear_decay_lr(progress_remaining):
    return 0.00027232300584036946 * progress_remaining
if __name__ == "__main__":
    vec_env = make_vec_env("ALE/Seaquest-ram-v5", n_envs=16)
    model = A2C(
        "MlpPolicy",
        vec_env,
        learning_rate=linear_decay_lr,
        n_steps=256,
        gamma=0.999,
        gae_lambda=0.98,
        ent_coef=0.00001753537605091099,
        vf_coef=0.19195701505334234,
        max_grad_norm=0.5,
        use_rms_prop=True,
        normalize_advantage=False,
        verbose=1,
        tensorboard_log="./seaquest/107",
        policy_kwargs=dict(activation_fn=torch.nn.Tanh, net_arch=dict(pi=[256, 256], vf=[256, 256]), ortho_init=True,
                                      optimizer_class=RMSpropTFLike, optimizer_kwargs=dict(eps=1e-5))
    )
    model.learn(total_timesteps=1000000, log_interval=1)
```
I get results:
![firefox_ETT9C6jsTI](https://github.com/DLR-RM/rl-baselines3-zoo/assets/148492076/8f2d80af-6ea4-4680-94da-e4720abf0553)

As picture shows, result is long way from 456 that RL Baselines Zoo got to. I have used more hyperparameters, but scores are always much lower.
What I'm aware of that can have impact on this issue is seed, as I didn't pick the same. Nevertheless I have tried many instances of A2C and the problem remains. 

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/rl-baselines3-zoo/issues) in the repo
- [X] I have read the [SB3 documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have read the [RL Zoo documentation](https://rl-baselines3-zoo.readthedocs.io)
- [X] If code there is, it is [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014)
- [X] If code there is, it is formatted using the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Probably a duplicate of https://github.com/DLR-RM/rl-baselines3-zoo/issues/314 https://github.com/DLR-RM/rl-baselines3-zoo/issues/204 and others (see link in the other issues)
